cJSON.c

```diff
@@ -935,7 +941,7 @@ static parse_buffer *buffer_skip_whitespace(parse_buffer * const buffer)
 /* Parse an object - create a new root, and populate. */
 CJSON_PUBLIC(cJSON *) cJSON_ParseWithOpts(const char *value, const char **return_parse_end, cJSON_bool require_null_terminated)
 {
-    parse_buffer buffer;
+    parse_buffer buffer = { 0 };
```

Comment:
But it's never accessed if `value` is `NULL`. Whatever I still want the code to compile cleanly with major compilers, even if there is a false positive.

I still wonder about the `{ 0 }` initializer. I that one is valid C89 and equivalent to setting all elements to `0`, then OK, otherwise memset is acceptable, as long as it gets optimized out.